tests/integration/test_bundle.py

```diff
@@ -276,6 +278,35 @@ async def test_loki_receives_logs(ops_test: OpsTest):
     logger.info("loki is successfully receiving flog logs")
 
 
+@pytest.mark.abort_on_fail
+async def test_all_traefik_proxied_urls_return_200(ops_test: OpsTest):
+    """Verify that all ingressed applications' URLs are reachable from localhost and return 200."""
+    proxied_urls = await get_all_proxied_urls(ops_test)
+    for app, url in proxied_urls.items():
+        response = urlopen(url, context=insecure_context)
+        assert response.code == 200
+
+
+@pytest.mark.abort_on_fail
+async def test_grafana_login_works(ops_test: OpsTest):
```

Comment:
We want to unify all the tests inside one repo, hence the main motivation to add these tests here.

Reply:
> unify all the tests inside one repo

So far we intentionally avoided doing so.
The traefik test introduced in this PR is a bundle concern, but logging in into grafana should be tested much earlier in the quality process.

In the context of a bundle test, I think it would be legitimate for a test to fail early if e.g. `curl --user admin:password http://IP_ADDRESS:3000/api/datasources/` fails auth.